server/test/db.test.ts

```diff
@@ -613,3 +614,43 @@ describe("db.getCurrentAvailabilityForLocations", () => {
     });
   });
 });
+
+describe("db.addExternalIds", () => {
+  const updatedData = {
+    external_ids: [
+      ["good", "1234"],
+      ["bad:", "987"],
+    ],
+  };
+
+  describe("creating locations", () => {
+    it("does not allow a colon in the externalId", async () => {
+      const TestLocationInvalidExternalId = {
+        ...TestLocation,
+        ...updatedData,
+      };
+      await expect(async () => {
+        await createLocation(TestLocationInvalidExternalId);
+      }).rejects.toThrow(ValueError);
+
+      const { external_ids } = await getLocationById(
+        TestLocationInvalidExternalId.id
+      );
+      expect(external_ids).toEqual([]);
```

Comment:
> If `:` is present on create, the provider location is created without any external ids. If present on update, the original external Ids are preserved.

So that shouldn’t actually happen; the transaction wrapper here should cause the location not to be created at all: https://github.com/usdigitalresponse/univaf/blob/25a12f1cfb717a4bf5e57e53a0ddc8bd7d4e134c/server/src/db.ts#L102-L115

(And that seems to work when I test this PR manually.)

I’m pretty sure what’s happening in this test is that because the whole test itself is wrapped in a transaction ([that’s what `installTestDatabaseHooks()` does](https://github.com/usdigitalresponse/univaf/blob/25a12f1cfb717a4bf5e57e53a0ddc8bd7d4e134c/server/test/lib.ts#L65-L70)), the inner transaction in `createLocation` is effectively a no-op. I wonder if we need an alternative version of `installTestDatabaseHooks()` that truncates tables instead… not sure that’s critically necessary for this PR, though it would be nice.